extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/DidJwtCredentialsVerifier.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.identityhub.verifier;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import com.nimbusds.jwt.proc.BadJWTException;
+import com.nimbusds.jwt.proc.DefaultJWTClaimsVerifier;
+import org.eclipse.dataspaceconnector.iam.did.spi.key.PublicKeyWrapper;
+import org.eclipse.dataspaceconnector.iam.did.spi.resolution.DidPublicKeyResolver;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.text.ParseException;
+import java.util.Set;
+
+/**
+ * Verifies verifiable credentials in JWT format with a DID issuer.
+ */
+class DidJwtCredentialsVerifier implements JwtCredentialsVerifier {
+
+    private final DidPublicKeyResolver didPublicKeyResolver;
+    private final Monitor monitor;
+
+    // RFC 7519 Registered (standard) claim
+    private static final String ISSUER_CLAIM = "iss";
+
+    DidJwtCredentialsVerifier(DidPublicKeyResolver didPublicKeyResolver, Monitor monitor) {
+        this.didPublicKeyResolver = didPublicKeyResolver;
+        this.monitor = monitor;
+    }
+
+    public boolean isSignedByIssuer(SignedJWT jwt) {
+        String issuer;
+        try {
+            issuer = jwt.getJWTClaimsSet().getIssuer();
+        } catch (ParseException e) {
+            monitor.warning("Error parsing issuer from JWT", e);
+            return false;
+        }
+        var issuerPublicKey = didPublicKeyResolver.resolvePublicKey(issuer);
+        if (issuerPublicKey.failed()) {
+            monitor.warning(String.format("Failed finding publicKey of issuer: %s", issuer));
+            return false;
+        }
+        Result<Void> signatureResult = verifySignature(jwt, issuerPublicKey.getContent());
+        return signatureResult.succeeded();
+    }
+
+    /**
+     * Verifies if a DID JWT targets the given subject.
+     *
+     * @param jwt             to be verified.
+     * @param expectedSubject subject claim to verify.
+     * @return if the JWT is for the given subject and signed by the claimed issuer.
+     */
```

Comment:
javadoc can be removed (as identical with interface, and for symmetry with other method)